### PR TITLE
Create User Story 18 - Edit User Profile

### LINKED ISSUE #59: User Story 18, User Can Edit their Profile Data
```
As a registered user
When I visit my profile page
I see a link to edit my profile data
When I click on the link to edit my profile data
Then my current URI route is "/profile/edit"
I see a form like the registration page
The form contains all of my user information
The password fields are blank and can be left blank
I can change any or all of the information
When I submit the form
Then I am returned to my profile page
And I see a flash message telling me that my data is updated
And I see my updated information
```

DO THESE STORIES FIRST:
- User Story 17, User Profile Show Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #61

child of #82

depends on #60

### PR DESCRIPTION
**Pull request by blake-enyart**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - 100%

# What functionality does this accomplish?
Creates edit functionality for a user on their profile page

# What did you struggle on to complete?
User of bcrypt has_secure_password validations overlapping responsibility with validates presence of password in model.

# Implements/Fixes:
* description
closes #59 

# Testing Changes
- [] No Tests have been changed
- [X] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [X] My code has no unused/commented out code
- [X] I have reviewed my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have fully tested my code

# Please include an emoji of how you feel about this branch:
💢

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -25,9 +26,27 @@ def create
...
+    redirect_to profile_path
+  end
+
   private
 
   def user_params
     params.require(:user).permit(:name, :street, :city, :state, :zipcode, :email, :password)
   end
+
+  def update_params
```
Comment (milevy1) [inline app/controllers/users_controller.rb]: @blake and @n-flint, since this always deletes the password, a user cannot ever change their password with this form.

Comment (blake-enyart) [reply app/controllers/users_controller.rb]: Corrected password edit page
